src/libraries/System.Net.Http.Json/tests/FunctionalTests/JsonContentTests.cs

```diff
@@ -0,0 +1,207 @@
+            JsonContent jsonContent2 = JsonContent.Create<object>(null);
+
+            jsonContent1.Headers.ContentType.CharSet = "foo-bar";
+
+            Assert.NotEqual(jsonContent1.Headers.ContentType.CharSet, jsonContent2.Headers.ContentType.CharSet);
+            Assert.NotSame(jsonContent1.Headers.ContentType, jsonContent2.Headers.ContentType);
+        }
+
+        [Fact]
+        public async Task JsonContentMediaTypeValidateOnServerAsync()
+        {
+            await LoopbackServer.CreateClientAndServerAsync(
+                async uri =>
+                {
+                    using (HttpClient client = new HttpClient())
+                    {
+                        var request = new HttpRequestMessage(HttpMethod.Post, uri);
+                        MediaTypeHeaderValue mediaType = MediaTypeHeaderValue.Parse("foo/bar; charset=utf-8");
+                        request.Content = JsonContent.Create(Person.Create(), mediaType: mediaType);
+                        await client.SendAsync(request);
+                    }
+                },
+                async server => {
+                    HttpRequestData req = await server.HandleRequestAsync();
+                    Assert.Equal("foo/bar; charset=utf-8", req.GetSingleHeaderValue("Content-Type"));
+                });
+        }
+
+        [Fact]
+        public void JsonContentMediaTypeDefaultIfNull()
+        {
+            Type fooType = typeof(Foo);
+            Foo foo = null;
+
+            JsonContent content = JsonContent.Create(foo, fooType, mediaType: null);
+            Assert.Equal("application/json", content.Headers.ContentType.MediaType);
+            Assert.Equal("utf-8", content.Headers.ContentType.CharSet);
+
+            content = JsonContent.Create(foo, mediaType: null);
+            Assert.Equal("application/json", content.Headers.ContentType.MediaType);
+            Assert.Equal("utf-8", content.Headers.ContentType.CharSet);
+        }
+
+        [Fact]
+        public void JsonContentInputTypeIsNull()
+        {
+            string foo = "test";
+
+            ArgumentNullException ex = Assert.Throws<ArgumentNullException>(() => JsonContent.Create(foo, inputType: null, mediaType: null));
+            Assert.Equal("inputType", ex.ParamName);
+        }
+
+        [Fact]
+        public void JsonContentThrowsOnIncompatibleTypeAsync()
+        {
+            HttpClient client = new HttpClient();
+            var foo = new Foo();
+            Type typeOfBar = typeof(Bar);
+
+            Exception ex = Assert.Throws<ArgumentException>(() => JsonContent.Create(foo, typeOfBar));
+
+            string strTypeOfBar = typeOfBar.ToString();
+            Assert.Contains(strTypeOfBar, ex.Message);
+
+            string afterInputTypeMessage = ex.Message.Split(strTypeOfBar.ToCharArray())[1];
+            Assert.Contains(afterInputTypeMessage, ex.Message);
+        }
+
+        [Fact]
+        public static async Task ValidateUtf16IsTranscodedAsync()
```

Comment:
This test only validates that the `Content-Type` header contains the expected value, not that the raw data sent on the wire is indeed UTF-16. Is this intentional?